lua/null-ls/info.lua

```diff
@@ -65,29 +78,82 @@ end
 M.show_window = function()
     local client = require("null-ls.client").get_client()
     local bufnr = api.nvim_get_current_buf()
+    local ft = api.nvim_buf_get_option(bufnr, "filetype")
     if not client or not lsp.buf_is_attached(bufnr, client.id) then
         log:warn("failed to get info: buffer is not attached")
         return
     end
 
+    local create_method_info = function(method)
+        local name = methods.readable[method]
+        local active_sources = {}
+        local supported_sources = {}
+
+        for _, source in ipairs(sources.get_available(ft, method) or {}) do
+            table.insert(active_sources, source.name)
+        end
+
+        -- the metadata is indexed by the builtin-names
+        for _, source in ipairs(sources.get_supported(ft, name) or {}) do
```

Comment:
```suggestion
        for _, source in ipairs(sources.get_supported(ft, method)) do
```

Wouldn't passing in `method` here let us keep the old readable names? This also will always return a table.

Reply:
I should've explained the problem better, but we index the generated metadata by the builtin-type, e.g. `formatting`
https://github.com/jose-elias-alvarez/null-ls.nvim/blob/2afe5c2e3484fc9e20589aae49e3829e0419596b/lua/null-ls/builtins/_meta/filetype_map.lua#L10-L13

So we have to actually either modify the generator script:
https://github.com/jose-elias-alvarez/null-ls.nvim/blob/2afe5c2e3484fc9e20589aae49e3829e0419596b/scripts/autogen_metadata.lua#L35-L38

or fix/add the mapping.

Reply:
Alright, I think I get it. I think we can follow the folder structure, then, meaning that diagnostics/diagnostics on save and formatting/formatting on save will be handled in the same way. We could then show more detail for each active source about its specific capabilities.